src/multilayerqg.jl

```diff
@@ -533,16 +535,50 @@ Compute the inverse Fourier transform of `varh` and store it in `var`.
 """
 invtransform!(var, varh, params::AbstractParams) = ldiv!(var, params.rfftplan, varh)
 
+"""
+    @kernel function pvfromstreamfunction_kernel!(qh, ψh, S, nlayers)
+
+Kernel for GPU acceleration of PV from streamfunction calculation, i.e., obtaining the Fourier
+transform of the PV from the streamfunction `ψh` in each layer using `qh = params.S * ψh`.
+"""
+@kernel function pvfromstreamfunction_kernel!(qh, ψh, S, nlayers)
```

Comment:
```suggestion
@kernel function pvfromstreamfunction_kernel!(qh, ψh, S, ::Val{nlayers}) where nlayers
```

for `@unroll` you have to do this, and also pass `Val(nlayers)` rather than `nlayers` into the kernel when launching it. I don't know if it will speed things up though. It might.